unpack.c

```diff
@@ -21,9 +21,10 @@ typedef struct {
 
 
 EMSCRIPTEN_KEEPALIVE
-ExtractedArchive* extract_archive(uint8_t* inputData, size_t inputSize, size_t* fileCount, bool decompressionOnly ) {
+ExtractedArchive* extract_archive(uint8_t* inputData, size_t inputSize, bool decompressionOnly ) {
     struct archive* archive;
     struct archive_entry* entry;
+    size_t files_struct_length = 100;
```

Comment:
This mimicks the behaviour std::vector for c++ uses as a strategies for allowing fast resizes without too frequent reallocation. Choosing 100 is quite random, the size of the struct is small and will not hurt.